### PR TITLE
fix(table data source): change default sort

### PR DESCRIPTION
**Pull request by corykon**
Sort like this 'AaBbCc' instead of this 'ABCabc'

re #661

### CODE REVIEW

File: projects/cashmere/src/lib/table/table.md [hunk 1]
```diff
@@ -129,6 +129,8 @@ If you are using the `HcTableDataSource` for your table's data source, provide t
 directive to the data source and it will automatically listen for sorting changes and change the
 order of data rendered by the table.
 
+The default sort order of values is as follows: "[Null]AaBbCc123".
```
Comment (joeskeen) [inline projects/cashmere/src/lib/table/table.md]: if you're doing `toLocaleLowerCase()`, isn't the sort order actually `ABC123` (case-insensitive)?  Wouldn't this be the sort order?
`aBc`
`AdE`

Comment (joeskeen) [reply projects/cashmere/src/lib/table/table.md]: Also, don't numbers usually sort before letters in alphabetical order?

Comment (corykon) [reply projects/cashmere/src/lib/table/table.md]: @joeskeen I'll make that change you suggested on the docs. The examples you're showing look correct to me. 

Yup typically numbers are sorted first, but the way the component is written they come second. Not my doing. I suppose we could change that too if we'd like

**Pull request merged**